app/javascript/middleware/new-provider-reducer.ts

```diff
@@ -0,0 +1,10 @@
+import { AppState } from '../packs/miq-redux';
+
+const asdf = 'INIT_NEW_PROVIDER_HAWKULAR'
+function initNewProvider(state, action): AppState {
```

Comment:
nitpick - can be written as
```js
const initNewProvider = (state, action): AppState => (
  {...state, data: action.payload.data}
);
```

see https://egghead.io/lessons/javascript-redux-simplifying-the-arrow-functions